### PR TITLE
Fixes from LGTM (and some code coverage)

### PR DESCRIPTION
**Pull request by oscargus**
#### Brief description of what is fixed or changed
Fixed a number of issues pointed out by LGTM.

As part of that, better testing of `Monomial` was introduced.

Furthermore, an example was fixed to work for Python 3.

#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/core/assumptions.py [hunk 1]
```diff
@@ -210,9 +210,8 @@ class StdFactKB(FactKB):
 
     This is the only kind of FactKB that Basic objects should use.
     """
-    rules = _assume_rules
```
Comment (oscarbenjamin) [inline sympy/core/assumptions.py]: Do you know what this line is/was supposed to do?

File: sympy/core/assumptions.py [hunk 2]
```diff
@@ -210,9 +210,8 @@ class StdFactKB(FactKB):
...
-
     def __init__(self, facts=None):
+        super(StdFactKB, self).__init__(_assume_rules)
```
Comment (oscarbenjamin) [inline sympy/core/assumptions.py]: Is this replacing the line removed above?

File: sympy/physics/quantum/matrixutils.py [hunk 3]
```diff
@@ -3,7 +3,6 @@
 from __future__ import print_function, division
 
 from sympy import MatrixBase, I, Expr, Integer
-from sympy.core.compatibility import range
```
Comment (oscarbenjamin) [inline sympy/physics/quantum/matrixutils.py]: I assume that this would give xrange instead of range under Python 2.

**Comment by oscarbenjamin** [comment]
I had a couple of questions/comments. Otherwise looks good.

File: sympy/core/assumptions.py (hunk 2 above)
Comment (oscargus) [reply sympy/core/assumptions.py]: Yes, it makes sense to call the super parent init so it results in the same result as setting rules above.

File: sympy/physics/quantum/matrixutils.py (hunk 3 above)
Comment (oscargus) [reply sympy/physics/quantum/matrixutils.py]: Maybe. `range` is not used in the file so that was the main motivation to remove it.

Comment (oscarbenjamin) [reply sympy/physics/quantum/matrixutils.py]: Okay!

**Comment by oscarbenjamin** [comment]
Thanks!

**Pull request merged**